server/src/main/java/com/cloud/network/IpAddressManagerImpl.java

```diff
@@ -304,7 +304,7 @@
 
     static Boolean rulesContinueOnErrFlag = true;
 
-    private static final ConfigKey<Boolean> SystemVmPublicIpReservationModeStrictness = new ConfigKey<Boolean>("Advanced",
+    public static ConfigKey<Boolean> SystemVmPublicIpReservationModeStrictness = new ConfigKey<Boolean>("Advanced",
```

Comment:
shouldn't this remain final? we don't want a global setting to get another config key in runtime. This has become more important as the var is now public.

Reply:
@DaanHoogland thanks for the suggestion.

I made these changes so that I could perform the unit tests. Unfortunately it was the only way I could test different configuration values. If you have any suggestions on this, I would be very grateful.

Reply:
I don't undestand why you would have to do that. The `IpAddressManager` you added for your test is already a mock and you can let it return anything you want. if need be you can add a `IsSVMReservationStrict()` method for convenience.

Reply:
Done, thanks

Reply:
> Done, thanks

I still see it as public, not private static @SadiJr .

Reply:
Done, thanks for the suggestion and sorry for the delay in answer.